### PR TITLE
style: refactorings suggested by the gocritic linter

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:

Increases the legibility of some code.

The [`gocritic`](https://github.com/go-critic/go-critic) linter suggests a lot of refactorings I don't think we would want to pursue, so I didn't enable it in our golangci-lint configuration.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: test/e2e/engine/template.go [hunk 1]
```diff
@@ -206,17 +206,19 @@ func (e *Engine) GetWindowsTestImages() (*WindowsTestImages, error) {
 		return nil, errors.New("Can't guess a Windows version without Windows nodes in the cluster")
 	}
 
-	if strings.Contains(e.ExpandedDefinition.Properties.WindowsProfile.GetWindowsSku(), "1809") || strings.Contains(e.ExpandedDefinition.Properties.WindowsProfile.GetWindowsSku(), "2019") {
+	windowsSku := e.ExpandedDefinition.Properties.WindowsProfile.GetWindowsSku()
+	switch {
+	case strings.Contains(windowsSku, "1809"):
```
Comment (jackfrancis) [inline test/e2e/engine/template.go]: I don't think this is equivalent. Are we assuming the 1809 case will fall through to the 2019 case?

I think a preferable change would be to use a single func that can evaluate a string against a set of string values for equality.

Comment (mboersma) [reply test/e2e/engine/template.go]: I added the `fallthrough` I forgot. I can pursue your suggested refactoring if that's preferable.

Comment (tariq1890) [reply test/e2e/engine/template.go]: +1 to jack's comment.

**Comment by tariq1890** [comment]
Thanks for this @mboersma !

File: pkg/api/agentPoolOnlyApi/v20180331/validate.go [hunk 2]
```diff
@@ -262,26 +262,20 @@ func validateAgentPoolVNET(a []*AgentPoolProfile) error {
 
 			if subscription == "" {
 				subscription = subnetSubscription
-			} else {
```
Comment (CecileRobertMichon) [inline pkg/api/agentPoolOnlyApi/v20180331/validate.go]: gotta love those else if statements

File: test/e2e/engine/template.go (hunk 1 above)
Comment (CecileRobertMichon) [reply test/e2e/engine/template.go]: can't we do  `case strings.Contains(windowsSku, "1809"), strings.Contains(windowsSku, "2019"):` if the two cases are the same result?

Comment (mboersma) [reply test/e2e/engine/template.go]: We can, and I think that reads better than using `fallthrough` anyway.

Comment (mboersma) [reply test/e2e/engine/template.go]: @jackfrancis @tariq1890 are you ok with the current changes?

**Pull request merged**